File: 189-rotate-array/189-rotate-array.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {void} Do not return anything, modify nums in-place instead.
 */

var reverse = (nums, start, end)=>{
        start = start
        end = end
        let temp = 0
        while (start < end){
            temp = nums[start]
            nums[start] = nums[end]
            nums[end] = temp
            start += 1
            end -= 1
        }
};
var rotate = function(nums, k) {
    k = k % nums.length
    if (nums.length == 1 || k == 0){return nums};
    reverse(nums, 0, nums.length - 1)
    reverse(nums, start=0, end=k-1)
    reverse(nums, start=k, end=nums.length-1)
    return nums
};

File: 189-rotate-array/189-rotate-array.py
class Solution(object):
    def reverse(self, nums, start, end):
        start = start
        end = end
        temp = 0
        while start < end:
            temp = nums[start]
            nums[start] = nums[end]
            nums[end] = temp
            start += 1
            end -= 1
            
    def rotate(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: None Do not return anything, modify nums in-place instead.
        """
        k = k % len(nums)
        
        self.reverse(nums, 0, len(nums) - 1)
        self.reverse(nums, start=0, end=k-1)
        self.reverse(nums, start=k, end=len(nums)-1)
        return nums
        